occ_poll_handler.cpp

```diff
@@ -0,0 +1,1259 @@
+                        "BYTE", lg2::hex, SensorName);
+                    break;
+            }
+
+            if (push_power_to_dbus == true)
+            {
+                // For Power field, Convert last 4 bytes into number value
+                SensorValue = (static_cast<uint32_t>(PollRspData[index + 10]) << 8) |
+                    (static_cast<uint32_t>(PollRspData[index + 11]));
+
+                // Convert output/DC power to input/AC power in Watts (round up)
+                uint16_t extnSensorValue =
+                    std::round(((SensorValue / (PS_DERATING_FACTOR / 100.0))));
+
+                dbus::OccDBusSensors::getOccDBus().setUnit(sensorPath, "xyz.openbmc_project.Sensor.Value.Unit.Watts");
+                dbus::OccDBusSensors::getOccDBus().setValue(sensorPath, extnSensorValue );
+                dbus::OccDBusSensors::getOccDBus().setOperationalStatus(sensorPath, !std::isnan(extnSensorValue));
+                if (statusObject.existingSensors.find(sensorPath) == statusObject.existingSensors.end())
+                {
+                    dbus::OccDBusSensors::getOccDBus().setChassisAssociation(
+                        sensorPath, {"all_sensors"});
+                }
+                statusObject.existingSensors[sensorPath] = occInstanceID;
+            }
+            index += bytesPerSensor;
+        }
+
+    }
+    else
+    {
+        if (TraceOncePollSensor)
+        {
+            lg2::error("OccPollHandler::PushExtnSensorsToDbus: unsuported format:{BYTE}",
+                "BYTE", lg2::hex, SensorFormat);
+            TraceOncePollSensor = false;
+        }
+
+        index += bytesPerSensor * NumberSensors;
+
+    }
+
+
+}//end OCCC PushExtnSensorsToDbus
+
+void OccPollHandler::PushExttSensorsToDbus(uint16_t& index )
+{
+    // Extn Sensor Header Data format for POLL
+    //  SensorFormat: 1 byte
+    //  bytesPerSensor: 1 byte
+    //  NumberSensors: 1 byte
+    //----------------------
+
+    uint8_t SensorFormat = PollRspData[index++];
+    uint8_t bytesPerSensor = PollRspData[index++];
+    uint8_t NumberSensors = PollRspData[index++];
+
+
+    if (TraceOncePollSensor)
+    {
+        lg2::error("OccPollHandler::PushExttSensorsToDbus: unsuported format:{BYTE}",
+            "BYTE", lg2::hex, SensorFormat);
+        TraceOncePollSensor = false;
+    }
+
+    //Temp to jump over section until format data put in.
+    index += bytesPerSensor * NumberSensors;
+
+    // Extt PROPOSED Sensor Record Data format for POLL
+    //  SensorID:  4 byte
+    //  FruType:   1 byte
+    //  Value:     1 byte
+    //----------------------
+
+}//end OCCC PushExttSensorsToDbus
+
+bool OccPollHandler::pollReadPcapBounds(uint32_t& capSoftMin, uint32_t& capHardMin, uint32_t& capMax)
+{
+    bool parmsChanged = true;
+
+    HandlePollAction();
+
+    capSoftMin = PollRspSoftMin;
+    capHardMin = PollRspHardMin;
+    capMax = PollRspMaxPower;
+
+    return parmsChanged;
+} //end OCCC pollReadPcapBounds
+
+
+#else
```

Comment:
Maybe you could put these in 2 different files?  It is kind of confusing with both functions in the same file (when searching, etc)
maybe occ_poll_handler_kernel.cpp ???

Did you give up on having a poll handler base class, and then have app / kernel objects?  or this is just first step?
```
Class OccPollHandler
{
    common vars/functions
}
Class AppPoll : public OccPollHandler
{
    app poll specific vars/functions
}
Class KernelPoll : public OccPollHandler
{
    kernel poll specific vars/functions
}

// Then use with:
OccPollHandler *pollHandler;
if (kernel)
{
    KernelPoll kPoll;
    pollHandler = &kPoll;
}
else
{
    AppPoll aPoll;
    pollHandler = &aPoll;
}

pollHandler->pollReadStateStatus();
...
```